supertokens_python/recipe/session/jwks.py

```diff
@@ -0,0 +1,101 @@
+import json
+import urllib.request
+from typing import List, Optional
+from urllib.error import URLError
+
+from jwt import PyJWK, PyJWKSet
+from jwt.api_jwt import decode_complete as decode_token  # type: ignore
+
+from supertokens_python.utils import get_timestamp_ms
+
+from .constants import JWKCacheMaxAgeInMs
+
+
+class JWKClient:
+    def __init__(
+        self,
+        uri: str,
+        cooldown_duration: int = 500,
+        cache_max_age: int = JWKCacheMaxAgeInMs,
+    ):
+        """A client for retrieving JSON Web Key Sets (JWKS) from a given URI.
+
+        Args:
+            uri (str): The URI of the JWKS.
+            cooldown_duration (int, optional): The cooldown duration in ms. Defaults to 500.
+            cache_max_age (int, optional): The cache max age in ms. Defaults to 300.
```

Comment:
This says ms, but it seems to be in seconds. I think using the same (ms) for both would be nice.